public/js/reducers/sources.js

```diff
@@ -13,15 +13,13 @@ import type { Record } from "../utils/makeRecord";
 export type SourcesState = {
   sources: I.Map<string, any>,
   selectedSource: ?any,
-  sourcesText: I.Map<string, any>,
   tabs: I.List<any>,
   sourceMaps: I.Map<string, any>,
 };
 
 const State = makeRecord(({
   sources: I.Map(),
   selectedSource: undefined,
-  sourcesText: I.Map(),
```

Comment:
Thanks for the writeup. Also, I apologize for leaving the `sourceText` change in the PR. It was not needed for pretty printing and should have been a separate PR / discussion. I'm going to take it out of this PR and open a second PR where we can have a discussion when you're back.

I think I hear you saying that you want the shape of the store to reflect the protocol data. e.g. separate message "new source", "source text", "loaded object" are separate fields somewhat like tables. One of the main benefits is ease of testing w/ the protocol data and actions. 

Personally, I'd like the shape of the store to be well shaped for the selectors / UI consumers. A couple of observations, we don't know what future clients we will support. It's possible we can overfit for one client and then the store is it's shape because of legacy (read support multiple languages). 

This conversation, prompted me to re-consider one part of your testing exploration. The tests are currently invoking actions like `newSource` with protocol data. The better approach would be to invoke client events with protocol data.  Here are two examples:

current action approach:
```js
store.dispatch(newSource(fakeSources["backbone.js"]));
```

preferred client event approach:
```js
const events = require("public/js/clients/firefox/events");
setupEvents({actions: boundActions, threadClient: () => {}});

events.newSource(undefined, fakeSources["backbone.js"]);
```

There are a couple of reasons why I think the client is the right place to mock:

The client transforms the protocol data into a form for the actions/store. 
+ Take `newSource` for example, it will set `isPrettyPrinted` to false and ignore some URLs. 
+ It's very likely that tests will fail because actions like `newSource` don't expect un-transformed protocol data
+ If we start changing the fixture data to be consistent with how the client transforms the data, there could be inconsistencies.

The client is a more natural place to mock commands. When we get to testing commands like `addBreakpoint`, the natural flow involves 
+ simulating a UI  `addBreakpoint` call with data that could come from the Editor
+ mocking the client, so that the correct protocol response is returned. The good news is that the protocol already involves request/response, so it should be possible to find the correct server response given request data.


Summary (that was a lot of words)
+ I'm dropping the sourceText change from this PR (should never have been here)
+ I'm opening a second PR w/ the sourceText change. I don't want to drag out the change, but I think it's an important discussion.
+ I'm going to add a comment on your testing proposal. This way we can have the right conversations in a focused way.